meta-balena-common/recipes-containers/hostapp-update/files/hostapp-update

```diff
@@ -37,19 +45,244 @@ run_current_hooks_and_recover () {
 	exit 1
 }
 
+# Test if a version is greater than another
+version_gt() {
+	test "$(echo "$@" | tr " " "\n" | sort -V | head -n 1)" != "$1"
+}
+
+# Dashboard progress helper
+progress() {
+	percentage=$1
+	message=$2
+	resin-device-progress --percentage "${percentage}" --state "${message}" > /dev/null || true
+}
+
+#######################################
+# Helper function to run a transient unit to update the supervisor.
+# Returns
+#  0: Success
+#  1: Failure
+#######################################
+_run_supervisor_update() {
+	local supervisor_update
+	local ret=0
+
+	supervisor_update="systemd-run --wait --unit run-update-supervisor update-balena-supervisor -n"
+	if ! eval "${supervisor_update}"; then
+		WARN "Supervisor couldn't be updated" && ret=1
+	fi
+	journalctl -a -u run-update-supervisor --no-pager || true
+	return "${ret}"
+}
+
```

Comment:
Running a transient systemd unit was required when the supervisor update was triggered remotely - there is no need to keep this complexity if it's the hostOS doing it.